### PR TITLE
Merge pull request #655 from jsvd/elasticsearch_version

### PR DESCRIPTION
**Pull request by robbavey**
new elasticsearch version handling

### CODE REVIEW

**Comment by jordansissel** [comment]
Waiting on travis, but LGTM if it is green.

**Comment by robbavey** [comment]
@jordansissel This has the same failures that the original commit had on master. I can either commit this as is and apply the fixes in a subsequent commit/backport (as per master), or I can fold those fixes into this commit to get the tests to pass straight away. WDYT?

**Comment by jsvd** [comment]
@robbavey I we can merge this as is, the failures are related to the lack of compatibility of the plugin with the es 6 wrt scripted updates, so we can work on those fixes in a separate issue/pr

File: lib/logstash/outputs/elasticsearch/template_manager.rb [hunk 1]
```diff
@@ -12,12 +12,10 @@ def self.install_template(plugin)
...
     private
-    def self.get_es_version(client)
-      client.get_version
-    end
-
     def self.get_es_major_version(client)
-      get_es_version(client)["number"][0]
+      # get the elasticsearch version of each node in the pool and
+      # pick the biggest major version
+      client.connected_es_versions.uniq.map {|version| version.split(".").first.to_i}.sort.last
```
Comment (jsvd) [inline lib/logstash/outputs/elasticsearch/template_manager.rb]: s/sort.last/max

there was already a PR in master to perform this change, I suggest doing it here too

**Pull request closed without merging**